### PR TITLE
refactor: Upgrade `connectivity_plus`

### PR DESCRIPTION
**Pull request by mbfakourii**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

## Issue

Closes: #972

## Approach
Upgrade `connectivity plus` to `5.0.1`

## Tasks

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, code comments)

### CODE REVIEW

**Comment by mtrezza** [comment]
You can close and re-open the PR to re-run the CI.

**Comment by mbfakourii** [comment]
> You can close and re-open the PR to re-run the CI. Why do you need to re-run the CI in the first place, do we have a flaky test?

In this [action](https://github.com/parse-community/Parse-SDK-Flutter/actions/runs/6547249394/job/17779352242), I encountered a codecov error and had to run CI again !

File: packages/flutter/pubspec.yaml [hunk 1]
```diff
@@ -31,7 +31,8 @@ dependencies:
   #   path: ../dart
 
   # Networking
-  connectivity_plus: ^3.0.6
+  connectivity_plus: ^5.0.1
+
```
Comment (mtrezza) [inline packages/flutter/pubspec.yaml:35]:
```suggestion
```

**Review by mtrezza — APPROVED** [review]
Looks good, this is only an internal package, so no breaking change?

The CI issue looks like a temporary service issue that self-resolved.

**Comment by mbfakourii** [comment]
> Looks good, this is only an internal package, so no breaking change?
> 
> The CI issue looks like a temporary service issue that self-resolved.

No, there is no problem, I think it is ready for merge.

**Comment by mbfakourii** [comment]
> If this should be released immediately then the `fix` PR requires a changelog entry. Otherwise you can make it a `refactor` to release it with the next `fix` or `feat` PR.

I think it is a `refactor` and there is no need for versioning

**Review by mtrezza — APPROVED** [review]

**Pull request merged**